solutions/USACO jan19-grass-planting.mdx

```diff
@@ -2,18 +2,20 @@
 id: usaco-894
 source: USACO Silver 2019 January
 title: Grass Planting
-author: Neo Wang
+author: Neo Wang, Aadit Ambadkar
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_planting_silver_jan19.html)
 
-<LanguageSection>
+## Solution 1 - Intended Solution
 
-<JavaSection>
+**Time Complexity:** $\mathcal{O}(N)$
 
-## Solution 1 (Java)
+Let $\texttt{adj}[i]$ represent the number of paths adjacent to node $i$ in the graph. Then, considering that node, and that node alone, you would need exactly $\texttt{adj}[i]+1$ numbers to fill it. This is because you would need $\texttt{adj}[i]$ for each of the adjacent nodes, and $1$ for the node itself. So, our answer is the maximum of all the adjacents, plus $1$. 
```

Comment:
```suggestion
Let $\texttt{deg}[i]$ represent the degree of node $i$: the number of paths connecting the node. Then, considering that node, and that node alone, you would need no less than $\texttt{deg}[i]+1$ grass types. This is because you would need $\texttt{deg}[i]$ for each of the adjacent nodes, and $1$ for the node itself. It can be shown that the tree can also be colored accordingly in $\max(\texttt{deg}[i])+1$ colors. 
```